src/main/groovy/com/lesfurets/jenkins/unit/BasePipelineTest.groovy

```diff
@@ -130,9 +130,20 @@ abstract class BasePipelineTest {
         helper.registerAllowedMethod("properties", [List])
         helper.registerAllowedMethod("pwd", [], { 'workspaceDirMocked' })
         helper.registerAllowedMethod("readFile", [String])
-        helper.registerAllowedMethod('retry', [Integer, Closure], { Integer count, Closure c ->
-            c.delegate = delegate
-            helper.callClosure(c)
+        helper.registerAllowedMethod("retry", [Integer, Closure], { Integer count, Closure c ->
+            def attempts = 0
+            while (attempts != count) {
```

Comment:
```suggestion
            while (attempts <= count) {
```

Paranoid, I know, but if someone called this mock with a negative value for `count`, then it would loop forever.